synthesizer/src/vm/verify.rs

```diff
@@ -76,6 +76,11 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
 
         /* Transaction */
 
+        // Ensure that the transaction is well formed and does not exceed the maximum size.
+        if let Err(error) = transaction.to_bytes_le() {
+            bail!("Transaction '{}' is not well-formed: {error}", transaction.id())
+        }
+
```

Comment:
This check should maybe also be in `atomic_speculate` as well to be extra sure?

Reply:
`check_transaction` is called before either instance of `atomic_speculate`. Is the redundancy necessary?